utils/keys_utils.go

```diff
@@ -0,0 +1,164 @@
+package utils
```

Comment:
I really don't link generating keys in the go layer. could you add a comment in the code explaining why aren't we able to use previously created keys? maybe we can refactor this latter